zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -3070,4 +3071,19 @@ private void validateACL(List<ACL> acl) throws KeeperException.InvalidACLExcepti
         }
     }
 
+    /**
+     * Returns comma separated list of users authenticated in the current session
+     *
+     * @return users
+     * @throws InterruptedException when interrupted
+     */
+    public String whoAmI() throws InterruptedException {
```

Comment:
I note the absence of an asynchronous version of the operation, but I suppose we can live without it :)

Reply:
What about returning a simple datastructure ?
in the future we will be able to return more data probably.

I would like to see something like:
class ClientInfo {
    String[] users;
    // auth scheme ?
    // other relevant info ?
}

Reply:
Good suggestion. 
Added  below structure
```
class ClientInfo {
        ustring authScheme; // Authentication scheme
        ustring user;       // user name or any other id(for example ip)
    }
```
Response changed to 
```
class WhoAmIResponse {
        vector<org.apache.zookeeper.data.ClientInfo> clientInfo;
    }
```

Reply:
> but I suppose we can live without it :)
yes, not required as of now, Can be added whenever there is use case.